test/shared/serialization/compositeSpecTests.ts

```diff
@@ -213,63 +332,309 @@ describe("compositeSpec", () => {
         }
       });
 
-      const recursiveCompositeSpec = compositeSpec("Letters", {
-        "A": {
-          valueSpec: stringSpec
-        },
-        "B": {
-          valueSpec: "Letters"
-        }
-      });
       compositeSerializeWithStrictTypeCheckingTest({
         testName: `should support recursive specs in JSON`,
-        propertySpecs: recursiveCompositeSpec.propertySpecs,
+        compositeSpec: tree,
         value: {
-          "A": "a",
-          "B": {
-            "B": {
-              "A": "aaa"
+          name: "A",
+          children: [
+            {
+              name: "B",
+              children: [
+                {
+                  name: "C"
+                }
+              ]
             }
-          }
+          ]
         },
         options: {
           compositeSpecDictionary: {
-            "Letters": recursiveCompositeSpec
+            "Tree": tree
           }
         },
         expectedResult: {
-          "A": "a",
-          "B": {
-            "B": {
-              "A": "aaa"
+          name: "A",
+          children: [
+            {
+              name: "B",
+              children: [
+                {
+                  name: "C"
+                }
+              ]
             }
-          }
+          ]
         }
       });
 
       compositeSerializeWithStrictTypeCheckingTest({
         testName: "should log and throw an error when a composite spec reference doesn't exist in composite spec dictionary",
-        propertySpecs: {
-          "A": {
-            valueSpec: "B"
+        compositeSpec: compositeSpec({
+          typeName: "Spam",
+          propertySpecs: {
+            "A": {
+              valueSpec: "B"
+            }
           }
-        },
+        }),
         value: {
           "A": "B doesn't exist in the composite TypeSpec dictionary"
         },
         expectedResult: new Error(`Missing composite specification entry in composite type dictionary for type named "B" at a.property.path.A.`),
         expectedLogs: [`ERROR: Missing composite specification entry in composite type dictionary for type named "B" at a.property.path.A.`]
       });
+
+      compositeSerializeWithStrictTypeCheckingTest({
+        testName: "should log and throw an error when a the composite spec is base type and value is missing property in base type",
```

Comment:
"when the composite spec..."

Reply:
Fixed in https://github.com/Azure/ms-rest-js/pull/63/commits/f63136568a28c87b9e35249566ad20090fe0064c.